platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1097,7 +1097,24 @@ - (id)mgl_jsonExpressionObject {
                 NSArray *arguments = self.arguments.mgl_jsonExpressionObject;
                 return [@[@"concat", self.operand.mgl_jsonExpressionObject] arrayByAddingObjectsFromArray:arguments];
             } else if ([function isEqualToString:@"objectFrom:withIndex:"]) {
-                return @[@"at", self.arguments[1].mgl_jsonExpressionObject, self.arguments[0].mgl_jsonExpressionObject];
+                id index = self.arguments[1].mgl_jsonExpressionObject;
+                
+                if ([self.arguments[1] expressionType] == NSConstantValueExpressionType
+                    && [[self.arguments[1] constantValue] isKindOfClass:[NSString class]]) {
+                    id value = self.arguments[1].constantValue;
+                    
+                    if ([value isEqualToString:@"FIRST"]) {
+                        index = [NSExpression expressionForConstantValue:@0].mgl_jsonExpressionObject;
+                    } else if ([value isEqualToString:@"LAST"]) {
+                        NSExpression *count = [NSExpression expressionWithFormat:@"count(%@) - 1", self.arguments[0]];
+                        id last = [count expressionValueWithObject:nil context:nil];
```

Comment:
What if the first argument is an expression like `CAST(color, 'NSArray')`? Casting a color to an array is unimplemented by default in NSExpression, so evaluating the expression will raise an exception. Instead, get the `mgl_jsonExpressionObject` of `count` itself, similar to how the `SIZE` case is implemented.